src/lsm/lsm_cursor.c

```diff
@@ -1558,14 +1558,13 @@ __clsm_remove(WT_CURSOR *cursor)
     __cursor_novalue(cursor);
     WT_ERR(__clsm_enter(clsm, false, true));
 
-    if (!F_ISSET(cursor, WT_CURSTD_OVERWRITE)) {
-        WT_ERR(__clsm_lookup(clsm, &value));
-        /*
-         * Copy the key out, since the insert resets non-primary chunk cursors which our lookup may
-         * have landed on.
-         */
-        WT_ERR(__cursor_needkey(cursor));
-    }
+    WT_ERR(__clsm_lookup(clsm, &value));
+    /*
+     * Copy the key out, since the insert resets non-primary chunk cursors which our lookup may have
+     * landed on.
+     */
+    WT_ERR(__cursor_needkey(cursor));
+
```

Comment:
Did this code fail some standalone test? How did you decide to change this?

Reply:
Yes, so there was a failure in `test_overwrite.py` where I expected that removing a non-existent record should fail in both cases where `overwrite=true` and `overwrite=false` but it actually succeeds.

Reply:
This is in line with normal cursors. Irrespective of `overwrite`, the remove operation fails if the key doesn't exist.

Reply:
@mm-ng , @kommiharibabu, unless I'm mistaken, I think the `overwrite` flag no longer has any meaning for `cursor.remove`, and I've removed all mention of it in 0c24f359a.